src/postgres/commands/PostgresAccountWizard/PostgresServerSetFirewallStep.ts

```diff
@@ -0,0 +1,44 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import PostgreSQLManagementClient from 'azure-arm-postgresql';
+import { FirewallRule } from 'azure-arm-postgresql/lib/models';
+import { Progress } from 'vscode';
+import { AzureWizardExecuteStep, createAzureClient } from 'vscode-azureextensionui';
+import { SubscriptionTreeItem } from '../../../tree/SubscriptionTreeItem';
+import { localize } from '../../../utils/localize';
+import { nonNullProp } from '../../../utils/nonNull';
+import { IPostgresWizardContext } from './IPostgresWizardContext';
+
+export class PostgresServerSetFirewallStep extends AzureWizardExecuteStep<IPostgresWizardContext> {
+    public priority: number = 250;
+
+    public async execute(wizardContext: IPostgresWizardContext, progress: Progress<{ message?: string; increment?: number }>): Promise<void> {
+
+        const ip: string = nonNullProp(wizardContext, 'publicIp');
+        const node: SubscriptionTreeItem = nonNullProp(wizardContext, 'subscriptonTreeItem');
+        const client: PostgreSQLManagementClient = createAzureClient(node.root, PostgreSQLManagementClient);
```

Comment:
You can use `wizardContext` instead of `node.root` here since it already has the subscription info necessary to create an azure client. Then you can remove `subscriptonTreeItem` from the wizard context